### ISSUE TITLE
[079] Run a controlled learning-rate experiment and report uncertainty

### ISSUE THREAD

**Issue by 25d-228** [25d-228/gpt-2-C#80]
**Prerequisite:** #78 and #1; optional BLAS issue #79 is not required. **Budget:** 35–55 minutes of planning/analysis; six training runs are separate. Learn research habits beyond making code execute.

## Fixed experiment
Create `docs/experiment-plan.md`, `tools/run_lr_experiment.sh`, `tools/summarize_experiment.py`, `tests/test_experiment_report.py`. Before running, write the hypothesis and primary metric: held-out selected-token mean cross-entropy after200 optimizer updates. Compare lr.001 versus.003 using seeds1234,1235,1236. For each seed, both rates start from the same freshly initialized weights—not from the previous run's checkpoint. Use scalar float, B1,T32,L1,H2,C16,V256, dropout0,decay0,clip1 and the authored train/validation dialogue data/masks from lesson077. Keep optimizer settings, batches, data order and update count identical. Do not use held-out text for gradients.

Run all six predeclared configurations, with evaluation every50 updates. Save raw train/validation CSV, exact CLI/config, data hashes, code commit, compiler/build/backend and initial seed. Validation CSV schema is `step,loss,perplexity,selected_tokens`; keep timing separate from deterministic numeric logs. Never select only successful-looking seeds or extend one condition until it wins.

The Python report uses the standard library only. For each learning rate report all three final held-out losses, their mean and **sample** standard deviation, plus train losses and any failures. Compare paired seeds descriptively; three tiny synthetic-data runs do not establish a universal best learning rate. An unexpected result is a valid finding, not a failed hypothesis to hide.

## Exact tests and commands
`tests/test_experiment_report.py`: values[1,2,3] have mean2 and sample SD1; reject a missing/duplicate seed, mixed data hashes/configs, nonfinite metrics, or a missing step200 record. Preserve raw CSV files and do not round before calculations.

Run `.venv/bin/python -m unittest discover -s tests -p 'test_experiment_report.py'`, `make test`, then `sh tools/run_lr_experiment.sh` and `.venv/bin/python tools/summarize_experiment.py --runs runs/lr-study --out docs/experiment-results.md`. The AI may supply explained orchestration/reporting helpers; training itself stays in C.

**Mutation:** omit the worst seed from a copied results folder; the completeness check must fail. Restore.

**Exit question:** What alternative explanation could account for your observed difference, and what experiment would distinguish it?

Record the preregistered hypothesis and actual conclusion in `notes/step-079.md`; follow #1.
